astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+        on these. An example Byte-By-Byte table is shown here.
+
+        See: http://vizier.u-strasbg.fr/doc/catstd-3.1.htx
+
+        Example::
+
+        --------------------------------------------------------------------------------
+        Byte-by-byte Description of file: table.dat
+        --------------------------------------------------------------------------------
+        Bytes Format Units  Label     Explanations
+        --------------------------------------------------------------------------------
+         1-  8  A8     ---    names   Description of names
+        10- 14  E5.1   ---    e       [-3160000.0/0.01] Description of e
+        16- 23  F8.5   ---    d       [22.25/27.25] Description of d
+        25- 31  E7.1   ---    s       [-9e+34/2.0] Description of s
+        33- 35  I3     ---    i       [-30/67] Description of i
+        37- 39  F3.1   ---    sameF   [5.0/5.0] Description of sameF
+        41- 42  I2     ---    sameI   [20] Description of sameI
+        44- 47  F4.1   h      RAh     Right Ascension (hour)
+        49- 51  F3.1   min    RAm     Right Ascension (minute)
+        53- 70  F18.15 s      RAs     Right Ascension (second)
+            72  A1     ---    DE-     Sign of Declination
+        73- 76  F5.1   deg    DEd     Declination (degree)
+        78- 81  F4.1   arcmin DEm     Declination (arcmin)
+        83- 98  F16.13 arcsec DEs     Declination (arcsec)
+
+        --------------------------------------------------------------------------------
+        """
+        # Get column widths
+        vals_list = []
+        col_str_iters = self.data.str_vals()
+        for vals in zip(*col_str_iters):
+            vals_list.append(vals)
+
+        for i, col in enumerate(self.cols):
+            col.width = max([len(vals[i]) for vals in vals_list])
+            if self.start_line is not None:
+                col.width = max(col.width, len(col.info.name))
+        widths = [col.width for col in self.cols]
+
+        startb = 1  # Byte count starts at 1.
+
+        # Set default width of the Bytes count column of the Byte-By-Byte table.
+        # This ``byte_count_width`` value helps align byte counts with respect
+        # to the hyphen using a format string.
+        byte_count_width = len(str(sum(widths) + len(self.cols) - 1))
+
+        # Format string for Start Byte and End Byte
+        singlebfmt = "{:" + str(byte_count_width) + "d}"
+        fmtb = singlebfmt + "-" + singlebfmt
+        # Add trailing single whitespaces to Bytes column for better visibility.
+        singlebfmt += " "
+        fmtb += " "
+
+        # Set default width of Label and Description Byte-By-Byte columns.
+        max_label_width, max_descrip_size = 7, 16
+
+        bbb = Table(names=['Bytes', 'Format', 'Units', 'Label', 'Explanations'],
+                    dtype=[str]*5)
+
+        # Iterate over the columns to write Byte-By-Byte rows.
+        for i, col in enumerate(self.cols):
```

Comment:
I think `i` is fine for a counting variable - it's what most people would expect when they see `i` or `j`. However, I don't feel strongly about this and would also be OK with a more explicit name like `col_num`.

Reply:
Using `i` is fine for a loop that can be easily seen in one screenful, while in this case the loop blocks are quite long.

While I'm looking at the code, the giant block in the `write` method could be done without any index variable by making a list `new_cols = []` at the top, then looping over columns and processing each column. For the special cases then some transformation of the column happens (possibly turning one column into two or more), but for the rest you just append the original column. Then at the end `self.cols = new_cols`. This way:
- You don't need an index
- You preserve the original column order. The current code always puts any coordinate columns at the end of the column list.

Reply:
Yup, I have also thought of doing this.
Since, the output of the tests will change because of different column sequence, it would be better to address this separately. So, **needs an issue**.